### PR TITLE
Update worker tablet type to drained in vtctld / docs

### PR DESCRIPTION
**Pull request by thompsonja**
Related to #2151.

Updated vtcombo to implement ChangeType so I could test it locally via run_local_databases

### CODE REVIEW

File: doc/VitessApi.md [hunk 1]
```diff
@@ -663,7 +663,7 @@ StreamExecute executes a streaming query based on shards. It depends on the quer
...
 | <code>BACKUP</code> | <code>6</code> | BACKUP is the type a server goes to when taking a backup. No queries can be served in BACKUP mode.  |
 | <code>RESTORE</code> | <code>7</code> | RESTORE is the type a server uses when restoring a backup, at startup time.  No queries can be served in RESTORE mode.  |
-| <code>WORKER</code> | <code>8</code> | WORKER is the type a server goes into when used by a vtworker process to perform an offline action. It is a serving type (as the vtworker processes may need queries to run). In this state, this tablet is dedicated to the vtworker process that uses it.  |
+| <code>DRAINED</code> | <code>8</code> | DRAINED is the type a server goes into when a process needs a separate tablet set aside for a specific action. For example, a vtworker process may set a tablet to drained to perform an offline action. It is a serving type (as the vtworker processes may need queries to run). In this state, this tablet is dedicated to the vtworker process that uses it.  |
```
Comment (pivanof) [inline doc/VitessApi.md]: s/vtworker process that uses it/background process that uses it/

File: doc/vtctlReference.md [hunk 2]
```diff
@@ -237,7 +237,7 @@ Makes the &lt;destination keyspace/shard&gt; serve the given type. This command
     * <code>schema_apply</code> &ndash; A slaved copy of data that had been serving query traffic but that is now applying a schema change. Following the change, the tablet will revert to its serving type.
     * <code>snapshot_source</code> &ndash; A slaved copy of data where mysqld is <b>not</b> running and where Vitess is serving data files to clone slaves. Use this command to enter this mode: <pre>vtctl Snapshot -server-mode ...</pre> Use this command to exit this mode: <pre>vtctl SnapshotSourceEnd ...</pre>
     * <code>spare</code> &ndash; A slaved copy of data that is ready but not serving query traffic. The data could be a potential master tablet.
-    * <code>worker</code> &ndash; A tablet that is in use by a vtworker process. The tablet is likely lagging in replication.
+    * <code>drained</code> &ndash; A tablet that is reserved for a process. For example, a tablet used by a vtworker process, where the tablet is likely lagging in replication.
```
Comment (pivanof) [inline doc/vtctlReference.md]: s/reserved for a process/reserved for a background process/

File: go/cmd/vtcombo/tablet_map.go [hunk 3]
```diff
@@ -617,7 +617,12 @@ func (itmc *internalTabletManagerClient) SetReadWrite(ctx context.Context, table
 }
 
 func (itmc *internalTabletManagerClient) ChangeType(ctx context.Context, tablet *topodatapb.Tablet, dbType topodatapb.TabletType) error {
-	return fmt.Errorf("not implemented in vtcombo")
+	t, ok := tabletMap[tablet.Alias.Uid]
+	if !ok {
+		return fmt.Errorf("tmclient: cannot find tablet %v", tablet.Alias.Uid)
+	}
+	t.agent.ChangeType(ctx, dbType)
```
Comment (pivanof) [inline go/cmd/vtcombo/tablet_map.go]: This change is not mentioned and not explained in the PR or commit description. Can you please add the explanation?

Comment (thompsonja) [inline go/cmd/vtcombo/tablet_map.go]: Done

File: doc/vtctlReference.md (hunk 2 above)
Comment (thompsonja) [inline doc/vtctlReference.md]: Done

**Comment by michael-berlin** [comment]
LGTM

**Pull request merged**